app/src/main/java/com/example/footballmanager/ui/screens/login/Login.kt

```diff
@@ -0,0 +1,174 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.ElevatedButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.sp
+import com.example.footballmanager.HomeActivity
+import com.example.footballmanager.R
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.firebase.Firebase
+import com.google.firebase.auth.AuthResult
+import com.google.firebase.auth.GoogleAuthProvider
+import com.google.firebase.auth.auth
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+
+@Composable
+fun LoginScreen() {
+    val ctx = LocalContext.current
+    var user by remember { mutableStateOf(Firebase.auth.currentUser) }
+    val putExtraName = stringResource(id = R.string.intent_putextra_user_name)
+    val launcher = rememberFirebaseAuthLauncher(
+        onAuthComplete = { result ->
+            user = result.user
+            val intent = Intent(ctx, HomeActivity::class.java)
+            intent.putExtra(putExtraName, user?.displayName.toString())
+            ctx.startActivity(intent)
+            val activity = ctx as Activity
+            activity.finish()
+        },
+        onAuthError = {
+            user = null
+        }
+    )
+    val token = stringResource(id = R.string.default_web_client_id)
+    val context = LocalContext.current
+
+    Column(
+        modifier = Modifier.fillMaxWidth(),
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        if (user == null) {
+            LogInOutButton(
+                onClickFunction = {
+                    val gso =
+                        GoogleSignInOptions
```

Comment:
Consider creating a manager dedicated to login with firebase operations. This would allow to delegate responsibility for creating a google login client or logout to another object